lib/pbench/server/api/resources/datasets_inventory.py

```diff
@@ -36,45 +34,45 @@ def __init__(self, config: PbenchServerConfig, logger: Logger):
                 API_METHOD.GET,
                 API_OPERATION.READ,
                 uri_schema=Schema(
-                    Parameter("dataset", ParamType.DATASET, required=True)
+                    Parameter("dataset", ParamType.DATASET, required=True),
+                    Parameter("path", ParamType.STRING, required=True),
                 ),
                 authorization=API_AUTHORIZATION.DATASET,
             ),
         )
 
-    def return_send_file(self, file_path):
-        return send_file(file_path)
-
-    def _get(self, params: ApiParams, request: Request) -> Response:
+    def _get(self, params: ApiParams, _) -> Response:
         """
-        Get the values of client-accessible dataset and retrun the byte stream of the requested file .
+        This function returns the contents of the requested file as a byte stream.
 
         Args:
-            params: Flask's URI parameter dictionary with dataset name
-            request: The original Request object containing query parameters
+            params: API parameters and it contains dataset and the path.
+
+        Raises:
+            UNSUPPORTED_MEDIA_TYPE error if the requested path isn't a file
+
 
         GET /api/v1/datasets/inventory/{dataset}/{path}
         """
 
         dataset = params.uri["dataset"]
         path = params.uri["path"]
 
-        # Validate the authenticated user's authorization for the combination
-        # of "owner" and "access".
-
-        self._check_authorization(
-            str(dataset.owner_id), dataset.access, API_OPERATION.READ
-        )
         try:
             file_tree = FileTree(self.config, self.logger)
-            dataset_location = file_tree.find_inventory(dataset.name)
-            file_path = Path(os.path.join(dataset_location, Path(path)))
-            if file_path.is_file():
-                return self.return_send_file(file_path)
-            else:
-                raise APIAbort(
-                    HTTPStatus.NOT_FOUND, "File is not present in the given path"
-                )
-
+            tarball = file_tree.find_dataset(dataset.name)
+            dataset_location = tarball.unpacked
         except DatasetNotFound as e:
             raise APIAbort(HTTPStatus.NOT_FOUND, str(e))
+        file_path = dataset_location / Path(path)
```

Comment:
> The second argument shouldn't be a Path, just a string.

The second argument to _what_?  (Nikhil seems to understand what you mean, @dbutenhof, but I'm not sure I'm following...I hope Siddardh is doing better than me.  😆)

Is `dataset_location` a `Path`??  (Gosh I miss having explicit types....)

If so, then I concur that it would be best not to convert `path` into a `Path`, as the `pathlib` rules for joining absolute paths can produce some counterintuitive results in certain situations (of which this might be one!).

Reply:
>Is dataset_location a Path??

Yeah @webbnh you're right :+1: , `dataset_location` will return a `Path` and the `path` variable is a string. In order to append these two, I'm converting the latter into `Path`[`pathlib`] for further references

Reply:
Right, @siddardh-ra, but you don't need to:  the [`/` operator when applied to a `Path`](https://docs.python.org/3/library/pathlib.html#operators) will accept a `str` for its other argument.  So, converting it to a `Path` first is likely a waste of time and space.